mm2src/coins/lp_coins.rs

```diff
@@ -3003,14 +3004,27 @@ where
 }
 
 #[cfg(target_arch = "wasm32")]
-fn save_history_to_file_impl<T>(coin: &T, ctx: &MmArc, history: Vec<TransactionDetails>) -> TxHistoryFut<()>
+fn save_history_to_file_impl<T>(coin: &T, ctx: &MmArc, mut history: Vec<TransactionDetails>) -> TxHistoryFut<()>
 where
     T: MmCoin + MarketCoinOps + ?Sized,
 {
     let ctx = ctx.clone();
     let ticker = coin.ticker().to_owned();
     let my_address = try_f!(coin.my_address().map_to_mm(TxHistoryError::InternalError));
 
+    // the transactions with block_height == 0 are the most recent so we need to separately handle them while sorting
+    history.sort_unstable_by(|a, b| {
+        if a.block_height == 0 {
+            Ordering::Less
+        } else if b.block_height == 0 {
+            Ordering::Greater
+        } else if a.block_height == b.block_height {
+            a.internal_id.cmp(&b.internal_id)
+        } else {
+            b.block_height.cmp(&a.block_height)
+        }
+    });
```

Comment:
First of all, I think this code snippet could be a separate function like it was `qrc20::history::sort_newest_to_oldest` as it's used in `not(target_arch = "wasm32")` and `target_arch = "wasm32"`.
Secondly, I think we could check if `a.block_height == b.block_height`:
```rust
if a.block_height == b.block_height {
    return a.internal_id.cmp(&b.internal_id);
}
// continue
```
It will help us to sort transactions with 0 height in only one way.

Reply:
```
  if a.block_height == b.block_height {
      a.internal_id.cmp(&b.internal_id)
  } else {
      b.block_height.cmp(&a.block_height)
  }
```

Can we use it this way? I didn't really understand the reason of handling 0 block heights separately.

Reply:
Yes, and we need to put the transactions with 0 height at the beginning (as most recent)
> the reason of handling 0 block heights separately.

But look:
```rust
if a.block_height == b.block_height {
    a.internal_id.cmp(&b.internal_id)
} else if a.block_height == 0 {
    Ordering::Less
} else if b.block_height == 0 {
    Ordering::Greater
} else {
    b.block_height.cmp(&a.block_height)
}
```
If `a.block_height == b.block_height == 0`, then we easily can compare their `internal_id`.
Otherwise we need to check if one of the transactions has 0 height or no transactions has 0 height.

Reply:
Okay now I get it